provisioning/device/samples/readme.md

```diff
@@ -0,0 +1,101 @@
+# Provisioning Device Client Sample - Microsoft Azure IoT SDK for .NET
+
+## Overview
+
+_This documentation is preliminary and subject to change._
+
+This folder contains samples demonstrating the steps required to dynamically associate devices with IoT Hubs using the Microsoft Azure IoT Device Provisioning Service.
+
+To ensure that only authorized devices can be provisioned, two device attestation mechanisms are supported by the service: one based on X509 certificates and another based on Trusted Platform Module (TPM) devices. In both cases, public/private key authentication will be performed during provisioning.
+
+_Preview only:_ The SDK currently supports two communication protocols: HTTP and AMQP over TLS.
+
+## Device provisioning in a nutshell
+
+Provisioning is achieved by using a single call to the `ProvisioningDeviceClient.RegisterAsync()` API specifying the IDScope (unique for each Provisioning Service deployment), a `SecurityClient` and a `ProvisioningTransportHandler`:
+
+```C#
+    ProvisioningDeviceClient provClient = ProvisioningDeviceClient.Create(s_idScope, security, transport);
+    DeviceRegistrationResult result = await provClient.RegisterAsync();
+    if (result.Status != ProvisioningRegistrationStatusType.Assigned) 
+    {
+    Console.WriteLine($"ProvisioningClient AssignedHub: {result.AssignedHub}; DeviceID: {result.DeviceId}");
+    }
+```
+
+### Provisioning devices using X509 certificate-based attestation
+
+Devices must have access to a single certificate with a private key. The private key can be hidden from the application using PCKS11 Hardware Security Modules.
+
+When an X509 certificate is used, both the _RegistrationID_ as well as the _DeviceID_ will be equal to the Common Name portion of the certificate Subject. (e.g. If the subject is `CN=mydevice O=Contoso C=US`, the RegistrationID and DeviceID will be `mydevice`.) The name must respect the [DeviceID naming constraints](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry).
+
+X509 attestation comes in two flavors: 
+
+1. Group Enrollment
```

Comment:
Please add details about how a device can use the derived certificate to provision via DPS as part of an enrollment group. This may need to be a sample using the entire cert chain as mentioned below somewhere.

Reply:
From the client's standpoint there is no difference: we only need the client certificate and its chain.

The X509 sample has this automated via the cert generation script for Individual Enrollment.
Developers already having certificates and their chains should follow instructions here:
https://github.com/Azure/azure-iot-sdk-csharp/pull/233/files#diff-9a7ac6d32ecad3deadba51e4b1fba932R43